cypress/utils/formFiller.ts

```diff
@@ -0,0 +1,272 @@
+      break;
+    case HANKE_KAISTAHAITTA.NELJA:
+      cy.get('#kaistaHaitta-item-3').click();
+      break;
+    case HANKE_KAISTAHAITTA.VIISI:
+      cy.get('#kaistaHaitta-item-4').click();
+      break;
+    default:
+      break;
+  }
+};
+
+export const selectKaistanPituusHaitta = (kaistanPituusHaitta: HANKE_KAISTAPITUUSHAITTA_KEY) => {
+  cy.get('#kaistaPituusHaitta-toggle-button').click();
+  switch (kaistanPituusHaitta) {
+    case HANKE_KAISTAPITUUSHAITTA.YKSI:
+      cy.get('#kaistaPituusHaitta-item-0').click();
+      break;
+    case HANKE_KAISTAPITUUSHAITTA.KAKSI:
+      cy.get('#kaistaPituusHaitta-item-1').click();
+      break;
+    case HANKE_KAISTAPITUUSHAITTA.KOLME:
+      cy.get('#kaistaPituusHaitta-item-2').click();
+      break;
+    case HANKE_KAISTAPITUUSHAITTA.NELJA:
+      cy.get('#kaistaPituusHaitta-item-3').click();
+      break;
+    case HANKE_KAISTAPITUUSHAITTA.VIISI:
+      cy.get('#kaistaPituusHaitta-item-4').click();
+      break;
+    default:
+      break;
+  }
+};
+
+export const selectMeluHaitta = (meluHaitta: HANKE_MELUHAITTA_KEY) => {
+  cy.get('#meluHaitta-toggle-button').click();
+  switch (meluHaitta) {
+    case HANKE_MELUHAITTA.YKSI:
+      cy.get('#meluHaitta-item-0').click();
+      break;
+    case HANKE_MELUHAITTA.KAKSI:
+      cy.get('#meluHaitta-item-1').click();
+      break;
+    case HANKE_MELUHAITTA.KOLME:
+      cy.get('#meluHaitta-item-2').click();
+      break;
+    default:
+      break;
+  }
+};
+
+export const selectPolyHaitta = (polyHaitta: HANKE_POLYHAITTA_KEY) => {
+  cy.get('#polyHaitta-toggle-button').click();
+  switch (polyHaitta) {
+    case HANKE_POLYHAITTA.YKSI:
+      cy.get('#polyHaitta-item-0').click();
+      break;
+    case HANKE_POLYHAITTA.KAKSI:
+      cy.get('#polyHaitta-item-1').click();
+      break;
+    case HANKE_POLYHAITTA.KOLME:
+      cy.get('#polyHaitta-item-2').click();
+      break;
+    default:
+      break;
+  }
+};
+
+export const selectTarinaHaitta = (tarinaHaitta: HANKE_TARINAHAITTA_KEY) => {
+  cy.get('#tarinaHaitta-toggle-button').click();
+  switch (tarinaHaitta) {
+    case HANKE_TARINAHAITTA.YKSI:
+      cy.get('#tarinaHaitta-item-0').click();
+      break;
+    case HANKE_TARINAHAITTA.KAKSI:
+      cy.get('#tarinaHaitta-item-1').click();
+      break;
+    case HANKE_TARINAHAITTA.KOLME:
+      cy.get('#tarinaHaitta-item-2').click();
+      break;
+    default:
+      break;
+  }
+};
+
+export const fillForm4 = (hankeData: HankeDataDraft) => {
+  typeof hankeData.haittaAlkuPvm === 'string'
+    ? cy.get('#haittaAlkuPvm').type(hankeData.haittaAlkuPvm)
+    : cy.get('#haittaAlkuPvm').type(hankeData.alkuPvm);
+
+  typeof hankeData.haittaLoppuPvm === 'string'
+    ? cy.get('#haittaLoppuPvm').type(hankeData.haittaLoppuPvm)
+    : cy.get('#haittaLoppuPvm').type(hankeData.loppuPvm);
+
+  cy.get('[data-testid=form4Header]').click(); // Close datepicker because it is over kaistaHaitta-toggle-button
+
+  if (typeof hankeData.kaistaHaitta === 'string') {
```

Comment:
Hmm, miksi tarkka tyyppitarkastus tässä? Eikö riitä if(hankeData.kaistaHaitta) ? 🤔

Reply:
Ahh, olin jossain vaiheessa käyttänyt "if key in object" -tyyppistä logiikkaa joka päästää läpi nullit -- ja apufunktiot ottivat pelkästään tiettyjä avainstringejä. Niin olin nostanut niiden poiston tuolle tasolle.
Siistin poies. 👍